src/Cli/dotnet/CommonOptions.cs

```diff
@@ -91,6 +95,18 @@ public static Argument DefaultToCurrentDirectory(this Argument arg)
                 "--interactive",
                 CommonLocalizableStrings.CommandInteractiveOptionDescription);
 
+        public static Option ArchitectureOption(bool includeShortVersion = true) =>
+            new ForwardedOption<string>(
+                includeShortVersion ? new string[] { "--arch", "-a" } : new string[] { "--arch" },
+                CommonLocalizableStrings.ArchitectureOptionDescription)
+            .SetForwardingFunction(ResolveArchOptionToRuntimeIdentifier);
+
+        public static Option OperatingSystemOption() =>
+            new ForwardedOption<string>(
+                "--os",
+                CommonLocalizableStrings.OperatingSystemOptionDescription)
```

Comment:
How about a proper long and short form for the Operating System option?

```suggestion
            new ForwardedOption<string>(
                includeShortVersion ? new string[] { "--operating-system", "-s" } : new string[] { "--operating-system" },
                CommonLocalizableStrings.OperatingSystemOptionDescription)
```

#### Notes

- If there's a way to add `--os` as an alias of the long form `--operating-system`, that would make this more friendly and intuitive!!!
- We could also do the shorter `--system` instead of the longer `--operating-system` but either is okay!

---

We could also call it as a platform but that's asking too much, since, platform means different in the MSBuild world!

```suggestion
            new ForwardedOption<string>(
                includeShortVersion ? new string[] { "--platform", "-p" } : new string[] { "--platform" },
                CommonLocalizableStrings.PlatformOptionDescription)
```

---

#### Examples

| Short Form | Long Form |
| --- | --- |
| `dotnet build -s win -a x64` | `dotnet build --os Windows --arch x64` |
| `dotnet build -s win -a x64` | `dotnet build --operating-system Windows --architecture x64` |
| `dotnet build -s win -a x64` | `dotnet build --system Windows --architecture x64` |
| `dotnet build -p win -a x64` | `dotnet build --platform Windows --architecture x64` |